code/src/modules/observations.cpp

```diff
@@ -11,36 +11,59 @@
 
 using namespace picopter;
 using namespace cv;
-Observations::Observations() {
 
+namespace picopter{
+
+Observations::Observations(Observation firstSighting) {
+
+    accumulator = firstSighting;
+    sightings.push_back(firstSighting);
+//    location = {Matx33d( 
+//        0, 0, 0,
+//        0, 0, 0,
+//        0, 0, 0),
+//        Vec3d(0,0,0)};
+//    velocity = stretchDistrib(generatedistrib(),0.1);   //0.1m/s uncertainty
+//    acceleration = stretchDistrib(generatedistrib(),0.1);   //0.1m/s/s uncertainty
+
+//    appendObservation(firstSighting);
 }
 
 
 //estimate the probability the given observation is of the same object
-double getSameProbability(Observation* observation){    
-
-return 0;
+//return the integral of  e^((x-l1).t()*A*(x-l1)) * e^((x-l2).t()*B*(x-l2))
+double Observations::getSameProbability(Observation observation){ 
+    Distrib C;
+    C.axes = (accumulator.location.axes.inv() + observation.location.axes.inv()).inv();
+    C.vect = accumulator.location.vect - observation.location.vect;
+    double retval = exp(-(((C.vect).t() * (C.axes * C.vect))(0)));
+return retval;
 }
 
 //add another sighting to this object
-void Observations::appendObservation(Observation* observation){     
+void Observations::appendObservation(Observation observation){
     sightings.push_back(observation);
-    //for(int i=0; i<_distrib_coeffs; i++){
-    //    distrib.coeffs[i] += observation->distrib.coeffs[i];
-    //}
-    location = combineDistribs(location, observation->location);
-    velocity = combineDistribs(velocity, observation->velocity);
-}
-void Observations::updateObject(){
-    double timestep = 1;
-    location = vectorSum(location, stretchDistrib(velocity, timestep));
-    velocity = vectorSum(velocity, stretchDistrib(acceleration, timestep));
+
+    accumulator.location = combineDistribs(accumulator.location, observation.location);
+    accumulator.velocity = combineDistribs(accumulator.velocity, observation.velocity);
+    //update the time stamp
+    if(accumulator.sample_time < observation.sample_time) accumulator.sample_time=observation.sample_time;
+    accumulator.source = observation.source;
+}
+void Observations::updateObject(TIME_TYPE timestep){
+
+    accumulator.location = vectorSum(accumulator.location, stretchDistrib(accumulator.velocity, ((double)(timestep.count()))/TICKS_PER_SEC ));
```

Comment:
Should probably be using `duration_cast<milliseconds>(timestep).count()` or similar